llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -21942,6 +21944,20 @@ SDValue tryLowerPartialReductionToDot(SDNode *N,
   else
     Opcode = AArch64ISD::UDOT;
 
+  // Partial reduction lowering for (nx)v16i8 to (nx)v4i64 requires an i32 dot
+  // product followed by a zero / sign extension
+  if ((ReducedType == MVT::nxv4i64 && MulSrcType == MVT::nxv16i8) ||
+      (ReducedType == MVT::v4i64 && MulSrcType == MVT::v16i8)) {
+    EVT ReducedTypeHalved =
+        (ReducedType.isScalableVector()) ? MVT::nxv4i32 : MVT::v4i32;
```

Comment:
(optional) nit:  `ReducedTypeHalved` -> `ReducedTypeI32`? 
To me `Halved` could mean half the vector width (e.g. v4 -> v2).

Reply:
Yes I agree, thank you for the advice, done.